plugins/confirm.cpp

```diff
@@ -306,45 +293,70 @@ class confirmation : public confirmation_base {
         return true;
     }
     bool feed (ikey_set *input) {
-        if (paused)
-        {
-            // we can only detect that we've left the screen by intercepting the
-            // ESC key
-            if (!paused_screen && input->count(df::interface_key::LEAVESCREEN))
-                conf_lua::api::unpause(NULL);
-            return false;
+        bool mouseExit = false;
+        if(df::global::enabler->mouse_rbut) {
+            mouseExit = true;
         }
-        else if (state == INACTIVE)
+        bool mouseSelect = false;
+        if(df::global::enabler->mouse_lbut) {
```

Comment:
should these be `...but` or `...but_down`? If the mouse is clicked, held, and then moved to the button, does it count?

Reply:
As soon as the mouse buttons are pushed down it counts, even if still holding the button. But it doesn't spam the events, so for example if you open trade, select some items and hold right click, the confirmation dialog appears and is not dismissed unless you let go of right click and click again.